### PR TITLE
docs: fix ISO language codes link to Sphinx language option link

### PR DESCRIPTION
**Pull request by tkoyama010**
The code used for sphinx translations is the language code, not the country code. ~Therefore, I fixed reference from [ISO 3166-1](https://ja.wikipedia.org/wiki/ISO_3166-1) to [IETF language tag](https://en.wikipedia.org/wiki/IETF_language_tag).~
Edit: It is complicated to see it in Wikipeida. So I change the link to sphinx document.

- [Sphinx document](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language).

Misconceptions of country codes and language codes are very common 🙂
Please read the excellent article [Why flags do not represent languages](https://www.flagsarenotlanguages.com/blog/why-flags-do-not-represent-language/) to find out why we need to separate the two 👍 

_Originally posted by @tkoyama010 in https://github.com/pyOpenSci/python-package-guide/issues/353#issuecomment-2265039618_

### CODE REVIEW

File: TRANSLATING.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ LANGUAGES = ["es", "fr"]
 ```
 
 ```{note}
-You can find a list of the two-letter ISO language codes [here](https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes).
+You can find a list of the two-letter IETF language tag [here](https://en.m.wikipedia.org/wiki/IETF_language_tag).
```
Comment (tkoyama010) [inline TRANSLATING.md:68]:
```suggestion
You can find a list of the two-letter IETF language tag [here](https://en.wikipedia.org/wiki/IETF_language_tag).
```

File: TRANSLATING.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ LANGUAGES = ["es", "fr"]
 ```
 
 ```{note}
-You can find a list of the two-letter ISO language codes [here](https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes).
+You can find a list of the two-letter IETF language tag [here](https://en.wikipedia.org/wiki/IETF_language_tag).
```
Comment (tkoyama010) [inline TRANSLATING.md:68]:
```suggestion
You can find a list of the two-letter sphinx language option [here](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language).
```

File: TRANSLATING.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ LANGUAGES = ["es", "fr"]
 ```
 
 ```{note}
-You can find a list of the two-letter ISO language codes [here](https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes).
+You can find a list of the two-letter sphinx language option [here](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language).
```
Comment (tkoyama010) [inline TRANSLATING.md:68]:
```suggestion
You can find a list of the two-letter Sphinx language option [here](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language).
```

**Review by willingc — APPROVED** [review]
Thanks @tkoyama010.

**Comment by lwasser** [comment]
it's so wonderful to have you here @tkoyama010 thank you for all of the work you are doing on translation!!

**Comment by tkoyama010** [comment]
@all-contributors please add @tkoyama010 for doc

**Comment by tkoyama010** [comment]
@all-contributors please add @willingc for review

**Pull request merged**